### PR TITLE
Improve documentation

### PR DESCRIPTION
**Pull request by vmapetr**
# Description
This PR creates `readme.md` with details about image-generation and deployment in `help` folder.

#### Related issue:
https://github.com/actions/virtual-environments/issues/564

## Check list
- [ ] Related issue / work item is attached
- [ ] Tests are written (if applicable)
- [ ] Documentation is updated (if applicable)
- [ ] Changes are tested and related VM images are successfully generated

### CODE REVIEW

File: help/CreateImageAndAzureResources.md [hunk 1]
```diff
@@ -1,11 +1,122 @@
-# Build an image.
+### General
+The virtual-environment project uses packer tool and contains templates for the following platforms: Windows 2016/2019, Ubuntu 16.04/18.04/20.04. 
+Each image template consists of a set of scripts that are executed on the virtual machine. 
+The Packer process initializes a connection to Azure subscription via Azure CLI, and automatically creates the temporary Azure resources required to build the source VM(temporary resource group, network interfaces, and VM from the "clean" image specified in the template). 
+If deployment succeeded, the build agent connects to the deployed VM and starts to execute installation steps from the selected template.
+In case if any template step fails, image generation will be aborted and temporary VM will be terminated.
+After successfull image generation, snapshot of temporary VM will be converted to VHD image and then uploaded to the specified Azure Storage Account.
```
Comment (maxim-lobanov) [inline help/CreateImageAndAzureResources.md:7]: typos:
`successful`
`a snapshot of the temporary VM`

File: help/CreateImageAndAzureResources.md [hunk 2]
```diff
@@ -1,11 +1,122 @@
...
+Where:
+- `SubscriptionId` - The Azure subscription Id where resources will be created.
+- `ResourceGroupName` - The Azure resource group name where the Azure virtual machine will be created.
+- `TemplateFilePath` - The path to the json ARM-template generated by packer during image generation locally.* 
+- `VirtualMachineName` - The name of the virtual machine to be generated.
+- `AdminUserName` - The administrator username for the virtual machine to be created.
+- `AdminPassword` - The administrator password for the virtual machine to be created.
+- `AzureLocation` - The location where the Azure virtual machine will be provisioned. Example: "eastus"
+
+\* *ARM-template can be obtained from the Packer output. For now, it seems like there is an [Az CLI bug](https://github.com/Azure/azure-cli/issues/5899) with specifying the template through a URI, so download the template from URI, that will be provided at the bottom of image-generation log, and use the local path of the template file.*
```
Comment (maxim-lobanov) [inline help/CreateImageAndAzureResources.md:82]: `that will be printed at the bottom`

File: help/CreateImageAndAzureResources.md [hunk 3]
```diff
@@ -1,11 +1,122 @@
...
+
+The function creates an Azure VM from a template and also generates network resources in Azure to make the VM accessible.
```
Comment (maxim-lobanov) [inline help/CreateImageAndAzureResources.md:84]: Usually, `also` is not used at the middle of sentense

File: help/CreateImageAndAzureResources.md (hunk 1 above)
Comment (vmapetr) [reply help/CreateImageAndAzureResources.md]: Thanks. Fixed.

File: help/CreateImageAndAzureResources.md [hunk 4]
```diff
@@ -1,11 +1,122 @@
-# Build an image.
+### General
```
Comment (maxim-lobanov) [inline help/CreateImageAndAzureResources.md:1]: I think we should change header structure. The root header should be `#`. Nested +1

File: help/CreateImageAndAzureResources.md (hunk 2 above)
Comment (vmapetr) [reply help/CreateImageAndAzureResources.md]: Thanks! Fixed.

File: help/CreateImageAndAzureResources.md (hunk 3 above)
Comment (vmapetr) [reply help/CreateImageAndAzureResources.md]: Thanks! Fixed.

File: help/CreateImageAndAzureResources.md (hunk 4 above)
Comment (vmapetr) [reply help/CreateImageAndAzureResources.md]: Yep.

**Pull request merged**